File: swift/02p2.swift
func getCommonCharStr(boxId1: String, boxId2: String) -> String {
  var commonCharStr = ""
  var diff = 0

#if swift(>=4.0)
  let indices = boxId1.indices
#elseif swift(>=3.0)
  let indices = boxId1.characters.indices
#endif

  for i in indices {
    if boxId1[i] == boxId2[i] {
      commonCharStr += String(boxId1[i])
    } else {
      diff += 1
    }
  }
  if diff == 1 { return commonCharStr }
  return ""
}

func getCorrectBoxCommonCharStr(boxIds: [String]) -> String {
  let boxCount = boxIds.count
  let jMax = boxCount - 1
  let iMax = jMax - 1
  for i in (0 ... iMax) {
    for j in (i + 1 ... jMax) {

      let str = getCommonCharStr(boxId1: boxIds[i], boxId2: boxIds[j])
      if str != "" { return str }
    }
  }
  return ""
}

var boxIds: [String] = [];
while let boxId = readLine() { boxIds.append(boxId) }

let str = getCorrectBoxCommonCharStr(boxIds: boxIds)
print("\(str)")

File: python/02p2.py
#!/usr/bin/env python

import sys

def get_common_char_str(box_id_1, box_id_2):
    common_chars = []
    diff = 0

    for i in range(0, len(box_id_1)):
        c = box_id_1[i : i + 1]
        if c == box_id_2[i : i + 1]:
            common_chars.append(c)
        else:
            diff += 1
    if diff == 1:
        return ''.join(common_chars)
    return ''

def get_correct_box_common_char_str(box_ids):
    box_count = len(box_ids)
    for i in range(0, box_count - 1):
        for j in range(i + 1, box_count):
            str = get_common_char_str(box_ids[i], box_ids[j])
            if str != '': return str
    return ''

box_ids = [box_id.rstrip() for box_id in sys.stdin]
print get_correct_box_common_char_str(box_ids)
